docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/AutoCounter.rst

```diff
@@ -95,19 +112,41 @@ By default, the read-rate is set to 0 cycles, which disables AutoCounter.
    readrate=100
 
 
-Upon setting this value, when you run a workload, an AutoCounter output file
-will be placed in the ``sim_slot_<slot #>`` directory on the F1 instance under
-the name ``AUTOCOUNTERFILE<N>``, with one file generated per clock domain
-containing an AutoCounter event. The header of each output file indicates the
-associated clock domain and its frequency relative to the base clock.
-
 .. Note:: AutoCounter is designed as a coarse-grained observability mechanism, as sampling 
       each counter requires two (blocking) MMIO reads (each read takes O(100) ns on EC2 F1).
       As a result sampling at intervals less than O(10000) cycles may adversely affect
       simulation performance for large numbers of counters.
-      If you intend on reading counters at a finer granularity, please consider using
+      If you intend on reading counters at a finer granularity, consider using
       synthesizable printfs.
 
+AutoCounter CSV Output Format
+---------------------------------
+AutoCounter output files are CSVs generated in the working directory where the
+simulator was invoked (this applies to metasimulators too), with the default
+names ``AUTOCOUNTERFILE<N>.csv``, one per clock domain. The CSV output format is
+depicted below, assuming a sampling period of ``N`` base clock cycles.
+
+.. csv-table:: AutoCounter CSV Format
+    :file: autocounter-csv-format.csv
+
+Column Notes:
+
+#. Each column beyond the first two corresponds to a PerfCounter instance in the clock domain.
+#. Column 0 past the header corresponds to the base clock cycle of the sample.
+#. The local cycle counter (column 1) is implemented as an always enabled
+   single-bit event, and increments even when the target is under reset.
+
+Row Notes:
```

Comment:
Perhaps a better call is to instead version the csv, and put a version string in the first line? I don't mind putting the header length at the top.

Reply:
Yeah.  Versioning it is maybe a good idea but it seemed a bit heavyweight.  However, if you keep good changelog for the format between versions it will make it easier to handle reading them later.